### ISSUE TITLE
OSError: all-MiniLM-L6-v2 is not a local folder and is not a valid model identifier listed on 'https://huggingface.co/models

### ISSUE THREAD

**Issue by mai1x9** [UKPLab/sentence-transformers#3106]
Hi

I have trying to use this library, but after running the model for 4 or 5 times, i started to get the error.
```
OSError: all-MiniLM-L6-v2 is not a local folder and is not a valid model identifier listed on 'https://huggingface.co/models'
If this is a private repository, make sure to pass a token having permission to this repo either by logging in with `huggingface-cli login` or by passing `token=<your_token>`
```

- Do i need to login into huggingface for it
- Or how to manually download the weights and use it.

**Comment by tomaarsen** [comment]
Hello!

I suspect that you were unable to connect to Hugging Face temporarily, for some reason. If this is common, then you can save the model to locally instead:
1. 
	a. `SentenceTransformer("all-MiniLM-L6-v2").save_pretrained("all-MiniLM-L6-v2-local")` if you can connect again, otherwise
    b. `git clone https://huggingface.co/sentence-transformers/all-MiniLM-L6-v2 all-MiniLM-L6-v2-local`
2. `SentenceTransformer("all-MiniLM-L6-v2-local")` should work as normal. I think you can also use `local_files_only=True` then, e.g.: `SentenceTransformer("all-MiniLM-L6-v2-local", local_files_only=True)`.

But normally, `SentenceTransformer("all-MiniLM-L6-v2")` should work just fine - it doesn't actually download the files every single time, everything is cached. The only reason that it still tries to find the model on Hugging Face is because it wants to check if there's an update that it should download.

- Tom Aarsen

**Comment by mai1x9** [comment]
> SentenceTransformer("all-MiniLM-L6-v2").save_pretrained("all-MiniLM-L6-v2-local")

Thank you very much, it indeed worked. I used hugging face download , it got some blobs, refs and snapshot with some hash, bit complicated.

- But the above code you provided did worked. Thank you.

**Comment by tomaarsen** [comment]
Indeed, that's our cache format, it's a bit more complex as most people usually don't need to interact with it manually :)

I'm glad you got it working!

- Tom Aarsen
